expression/builtin.go

```diff
@@ -173,9 +177,113 @@ func (b *baseBuiltinFunc) getArgs() []Expression {
 }
 
 func (b *baseBuiltinFunc) vecEval(input *chunk.Chunk, result *chunk.Column) error {
+	if b.self != nil && !b.enableVec {
+		return b.row2Vec(input, result)
+	}
 	return errors.Errorf("baseBuiltinFunc.vecEval() should never be called, please contact the TiDB team for help")
 }
 
+// row2Vec converts this built-in functions from row-based evaluation to vectorized evaluation.
+// It evaluates each row in input according to its row-based evaluation methods and updates the result.
+func (b *baseBuiltinFunc) row2Vec(input *chunk.Chunk, result *chunk.Column) error {
```

Comment:
how about modifying the function signature to:
```go
func row2Vec(f builtinFunc, input *chunk.Chunk, result *chunk.Column) error 
```

Reply:
But how can we get this builtinFunc `f`?